### PR TITLE
BEHAVIOR: Gets oracle approach to run on sorting books task!

### PR DESCRIPTION
**Pull request by NishanthJKumar**
Several changes to *finally* get our oracle SeSaMe approach to run on BEHAVIOR pick-and-place tasks (currently, just works on sorting books task, but should be straightforward to add new tasks).  

Summary of major changes:
* fixes the grasp sampler
* adds an error-correcting loop to movement of the hand within the grasp option model, leading to a much greater grasp success rate
* adjusts navigation sampler to better complement grasp sampler and lead to more grasp successes (by navigating closer to objects)
* changes state loading/saving to actually require file I/O. This (1) reduces memory usage and prevents our process from getting killed by the OS for using up too much RAM (which had been happening) and (2) allows perfect resets of the environment (including the state of all objects, whereas previously, some `URDFObjects` were being not reset, leading to planning failures

### CODE REVIEW

File: src/envs/behavior_options.py
Comment (tomsilver) [inline src/envs/behavior_options.py:555]: can the outer parentheses be removed?

File: src/envs/behavior_options.py
Comment (tomsilver) [inline src/envs/behavior_options.py:588]: i'm more used to seeing `isinstance(x, list)` than `isinstance(x, List)`, but I can't think of any reason to prefer the former... hopefully they're equivalent? i googled around for a minute but didn't learn anything \shrug

File: src/envs/behavior_options.py
Comment (tomsilver) [inline src/envs/behavior_options.py:1081]: warp speeds lol

**Review by tomsilver — COMMENTED** [review]
looking good!

File: .gitignore
Comment (ronuchit) [reply .gitignore:18]: note: see uses of "tempfile" in torch_models.py for a usage example

**Comment by ronuchit** [comment]
I'll review after Tom's feedback is incorporated. Btw, are there any updates/pulls we have to make on any other repo? I remember we're using some fork of igibson or something...

File: .gitignore
Comment (tomsilver) [reply .gitignore:18]: Actually instead of creating a temporary directory, maybe we can just use tempfile NamedTemporaryFile and that's it; there's no need for the files to be in any particular directory.

Comment (NishanthJKumar) [reply .gitignore:18]: sounds good! thanks for this suggestion!

**Comment by NishanthJKumar** [comment]
yeah - to run this code, you'd need to pull our fork of iGibson and BDDL.
I'll update the README with instructions for behavior in a separate PR

File: .gitignore
Comment (NishanthJKumar) [reply .gitignore:18]: i think we need a temp directory, since each `save_checkpoint` call saves two different files

File: src/envs/behavior.py
Comment (tomsilver) [inline src/envs/behavior.py:345]: I just took another look here and noticed the `str(simulator_state)`. what's going on there? isn't `simulate_state` like a really large and gross dictionary?

File: .gitignore
Comment (NishanthJKumar) [reply .gitignore:18]: i'm not sure how/when to automatically delete these directories though

Comment (tomsilver) [reply .gitignore:18]: cool. so then `tempfile.TemporaryDirectory` is probably what we want.

Comment (tomsilver) [reply .gitignore:18]: or maybe `dirpath = tempfile.mkdtemp()`. and btw, we might want to automatically delete these to avoid clogging up our storage. see: https://stackoverflow.com/questions/3223604/how-to-create-a-temporary-directory-and-get-its-path-file-name

Comment (tomsilver) [reply .gitignore:18]: haha I just saw your comment, GMTA! yeah let's see...

File: src/envs/behavior.py
Comment (NishanthJKumar) [reply src/envs/behavior.py:115]: I feel like there's not too much point in this, since it'll change 2 lines into 1 line, but we'll have to add a new class function that itself calls `load_checkpoint_states`, which is now its own separate function. I'm more inclined to keep things this way, but take a look at the new refactor and let me know if you still feel we should implement this change and I'm happy to do it!

Comment (tomsilver) [reply src/envs/behavior.py:120]: got it, makes sense. I'll defer to your judgement!

File: src/envs/behavior.py
Comment (ronuchit) [inline src/envs/behavior.py:29]: Use "exist_ok" argument instead of this if (see other usages in this repo for examples)

File: src/envs/behavior.py
Comment (ronuchit) [inline src/envs/behavior.py:477]: can we do `env.step(np.zeros_like(env.action_space.shape))`?

File: src/envs/behavior_options.py
Comment (ronuchit) [inline src/envs/behavior_options.py:247]: just curious, how come this argument isn't used? how can we make a navigation option without knowing what we're navigating to?

File: src/envs/behavior_options.py
Comment (ronuchit) [inline src/envs/behavior_options.py:564]: as in another comment i made, can we somehow use env.action_space here?

Comment (ronuchit) [reply src/envs/behavior_options.py:564]: just trying to reduce the number of magic numbers where possible

File: src/envs/behavior_options.py
Comment (ronuchit) [inline src/envs/behavior_options.py:662]: lol was this a bug before?

File: src/envs/behavior_options.py
Comment (ronuchit) [inline src/envs/behavior_options.py:654]: lol was this a bug before?

**Review by ronuchit — APPROVED** [review]
looks fine to me, minor comments but also wait for @tomsilver 's approval

File: src/envs/behavior_options.py
Comment (NishanthJKumar) [reply src/envs/behavior_options.py:247]: because we already have the plan that was computed by RRT!

File: src/envs/behavior_options.py
Comment (NishanthJKumar) [reply src/envs/behavior_options.py:662]: yup, and a pretty serious one lol

File: src/envs/behavior_options.py
Comment (NishanthJKumar) [reply src/envs/behavior_options.py:654]: yup, and it's why grasping wasn't working

File: src/envs/behavior_options.py
Comment (tomsilver) [inline src/envs/behavior_options.py:583]: here's another place where we could use `action_space` instead -- check all occurrences of 17

**Review by tomsilver — APPROVED** [review]
one tiny thing, otherwise lgtm, merge when ready!

File: src/envs/behavior_options.py
Comment (ronuchit) [reply src/envs/behavior_options.py:583]: `git grep 17` :D

**Pull request merged**
